File: abc/209/problem_c2.py
# -*- coding: utf-8 -*-

def main():
    n = int(input())
    c_list = [int(v) for v in input().split()]
    c_list.sort()

    ans = 1
    for i in range(n):
        # 順列の考え方で積算する。
        ans = ans * (c_list[i]-i)  # 一つ前より選択肢が一つ減る。
    print(ans % (10**9+7))


if __name__ == "__main__":
    main()

File: abc/209/problem_c2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;
    std::vector<unsigned long long> c_list(n);
    for (int i = 0; i < n; i++) {
        unsigned long long value;
        std::cin >> value;
        c_list[i] = value;
    }
    std::sort(c_list.begin(), c_list.end());;

    unsigned long long inf_val = (static_cast<unsigned long long>(1e9) + 7);
    unsigned long long ans = 1;
    for (int i = 0; i < n; i++) {
        // 順列の考え方で積算する。
        ans = ans * (c_list[i]-i);  // 一つ前より選択肢が一つ減る。
        ans = ans % inf_val;
        if (ans <= 0)
            break;
    }
    std::cout << ans << std::endl;
    return 0;
}
